src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -184,7 +186,36 @@ public function findByPid(string $pid, string $entityType = 'CoreEntity'): ?arra
      */
     public function findByPidFull(string $pid, string $entityType = 'CoreEntity'): ?array
     {
-        $qb = $this->findByPidFullCommon($pid, $entityType);
+        $qb = $this->findFullCommon($entityType);
+        $qb->andWhere('entity.pid = :pid')
```

Comment:
I prefer the $qb->expr()->eq(), if we have to use the QueryBuilder I think we should fully embrace it, don't you think?